### PR TITLE
Add table_type Hive property for CTAS queries

### PR DESCRIPTION
**Pull request by nezihyigitbasi**
Currently all tables created with Presto CTAS queries are managed tables. But we don't always want that as we don't want our data to be deleted on a `drop table`.  This PR adds support for creating external tables by adding a new Hive table property.

BTW currently `table_type` is passed directly to the metastore call, should we restrict the table type to only `MANAGED_TABLE` and `EXTERNAL_TABLE` (there are also `VIRTUAL_VIEW` and `INDEX_TABLE` types in `TableType` enum)?

### CODE REVIEW

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java [hunk 1]
```diff
@@ -469,12 +474,21 @@ else if (!partitionColumnNames.contains(name)) {
...
         if (sampled) {
             tableComment = "Sampled table created by Presto. Only query this table from Hive if you understand how Presto implements sampling.";
         }
-        table.setParameters(ImmutableMap.of("comment", tableComment));
+        ImmutableMap.Builder<String, String> tableParameters = ImmutableMap.builder();
+        tableParameters.put("comment", tableComment);
+
+        //Setting the table type to EXTERNAL_TABLE is not enough
+        //as Hive metastore also looks for the "EXTERNAL" property
+        if (hiveTableType.equals(EXTERNAL_TABLE)) {
```
Comment (nezihyigitbasi) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java]: Please see [this](https://github.com/apache/hive/blob/master/metastore/src/java/org/apache/hadoop/hive/metastore/MetaStoreUtils.java#[L1319-L1329]).

**Comment by martint** [comment]
Related: #3893

**Comment by electrum** [comment]
Sorry for the delay, I missed seeing this before.

First off, we definitely don't want to pass arbitrary types, so let's assume this is a boolean `external_table` flag. 

This seems like a strange feature. External tables seem to be meant reference data that is externally managed. Why would you want to create a table with data, but then leave the data orphaned in HDFS or S3 when the metadata is dropped?

If the intent is to protect against accidentally dropping the table, what if we support the Hive `NO_DROP` protection flag? We could have a table property to set it on creation, and a callable procedure to enable/disable (or directly change the table property if we decide to support altering properties of existing tables).

**Comment by nezihyigitbasi** [comment]
@electrum My intent was to protect the data from being dropped. I wasn't aware of the `NO_DROP ` flag and it sounds like a good start for that. [Btw this flag will be removed in Hive 2.0 (please see [this](https://issues.apache.org/jira/browse/HIVE-11145))]

For true support of external tables I guess we need to also have a location property in `create table`. 

For now I am closing this PR, if I have time I will take a look at the `NO_DROP` flag for drop protection.

**Comment by amrutagokhale** [comment]
@nezihyigitbasi Do you have any plans to restart work on this PR? Support for creating external tables in Presto would be helpful for us too, since that would simplify writing some of the tests for S3.

**Comment by nezihyigitbasi** [comment]
@amrutagokhale unfortunately no, you can work on it if you want to.

**Comment by robinup** [comment]
+1 need presto to be able to mount quickly rather than go through Hive

**Comment by electrum** [comment]
@amrutagokhale Can you explain the exact problem? Are you trying to create tables on S3? Or reference existing data on S3?

**Comment by amrutagokhale** [comment]
@electrum We wanted to create new tables in S3 for writing S3 tests. @rschlussel wrote these tests in here: [TestS3HiveConnector.java](https://github.com/Teradata/presto/blob/sprint-39/presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveS3Connector.java). Since Presto doesn't yet support creating external tables, the tests require that the tables have already been created, rather creating them on-the-fly.

**Pull request closed without merging**